core/src/main/java/com/linecorp/armeria/server/annotation/JavaBeanRequestConverterFunction.java

```diff
@@ -0,0 +1,181 @@
+package com.linecorp.armeria.server.annotation;
+
+import static com.linecorp.armeria.server.AnnotatedHttpServiceParamUtil.convertParameter;
+import static com.linecorp.armeria.server.AnnotatedHttpServiceParamUtil.httpParametersOf;
+import static org.reflections.ReflectionUtils.getAllFields;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.annotation.Nullable;
+
+import com.linecorp.armeria.common.AggregatedHttpMessage;
+import com.linecorp.armeria.common.HttpParameters;
+import com.linecorp.armeria.common.MediaType;
+import com.linecorp.armeria.server.AnnotatedHttpServiceParamUtil.EnumConverter;
+import com.linecorp.armeria.server.ServiceRequestContext;
+
+import io.netty.util.AsciiString;
+
+public class JavaBeanRequestConverterFunction implements RequestConverterFunction {
+    private final Map<Class<?>, BeanParamInfo> beanParamsMap = new HashMap<>();
+
+    @Override
+    public Object convertRequest(final ServiceRequestContext ctx,
+                                 final AggregatedHttpMessage request,
+                                 final Class<?> expectedResultType) throws Exception {
+        if (!shouldHandle(request)) {
+            return RequestConverterFunction.fallthrough();
+        }
+
+        final BeanParamInfo beanParamInfo =
+                beanParamsMap.computeIfAbsent(expectedResultType,
+                                              JavaBeanRequestConverterFunction::buildBeanParamInfo);
```

Comment:
Is there any way to build `BeanParamInfo` when the server starts up? I'm a little worried about the performance when many requests are coming at first time.

Reply:
It can be done, but need to add a new method like `prepare` or `init` to the `RequestConverterFunction` interface, and call this new method from `AnnotatedHttpServiceMethod` at the server starts up phase.
Does this strategy sounds good?

If you agree with it, I will do it.

Reply:
I think that it's enough to add a new method like `prepare` or `init` to the `JavaBeanRequestConverterFunction` only. 
If you create a singleton instance of that class in `AnnotatedHttpServices` class, we may build `BeanParamInfo` by calling `prepare` method of the singleton instance [here (maybe)](https://github.com/line/armeria/blob/master/core/src/main/java/com/linecorp/armeria/server/AnnotatedHttpServices.java#L449). Also, we may add the converter only if necessary. 
(It means that we can remove it from the default converter list)
What do you think?

Reply:
fixed.